samples/vboxwrapper/vbox_vboxmanage.cpp

```diff
@@ -58,821 +58,816 @@ using std::string;
-        if (retval) return retval;
+                // Deregister stale VM by UUID
+                return deregister_vm(false);
+            }
+        } else {
+            command  = "showhdinfo \"" + virtual_machine_slot_directory + "/" + image_filename + "\" ";
+            retval = vbm_popen(command, output, "get HDD info");
+            if (retval) return retval;
 
-        uuid_start = output.find("(UUID: ");
-        if (uuid_start != string::npos) {
-            // We can parse the virtual machine ID from the output
-            uuid_start += 7;
-            uuid_end = output.find(")", uuid_start);
-            vm_name = output.substr(uuid_start, uuid_end - uuid_start);
+            uuid_start = output.find("(UUID: ");
+            if (uuid_start != string::npos) {
+                // We can parse the virtual machine ID from the output
+                uuid_start += 7;
+                uuid_end = output.find(")", uuid_start);
+                vm_name = output.substr(uuid_start, uuid_end - uuid_start);
 
-            // Deregister stale VM by UUID
-            return deregister_vm(false);
-        } else {
-            // Did the user delete the VM in VirtualBox and not the medium?  If so,
-            // just remove the medium.
-            command  = "closemedium disk \"" + virtual_machine_slot_directory + "/" + image_filename + "\" ";
-            vbm_popen(command, output, "remove virtual disk", false, false);
-            if (enable_floppyio) {
-                command  = "closemedium floppy \"" + virtual_machine_slot_directory + "/" + floppy_image_filename + "\" ";
-                vbm_popen(command, output, "remove virtual floppy disk", false, false);
-            }
-            if (enable_isocontextualization) {
-                command  = "closemedium dvd \"" + virtual_machine_slot_directory + "/" + iso_image_filename + "\" ";
-                vbm_popen(command, output, "remove virtual ISO 9660 disk", false);
-                if (enable_cache_disk) {
-                    command  = "closemedium disk \"" + virtual_machine_slot_directory + "/" + cache_disk_filename + "\" ";
-                    vbm_popen(command, output, "remove virtual cache disk", false);
+                // Deregister stale VM by UUID
+                return deregister_vm(false);
+            } else {
+                // Did the user delete the VM in VirtualBox and not the medium?  If so,
+                // just remove the medium.
+                command  = "closemedium disk \"" + virtual_machine_slot_directory + "/" + image_filename + "\" ";
+                vbm_popen(command, output, "remove virtual disk", false, false);
+                if (enable_floppyio) {
+                    command  = "closemedium floppy \"" + virtual_machine_slot_directory + "/" + floppy_image_filename + "\" ";
+                    vbm_popen(command, output, "remove virtual floppy disk", false, false);
+                }
+                if (enable_isocontextualization) {
```

Comment:
This is code that can never be reached. The logic is:
```
if (enable_isocontextualization && enable_isocontextualization) {
    blah
} else {
    if (...) {
        blah
    } else {
        if (enable_isocontextualization) {
```

Reply:
Should I go ahead and delete completely this piece (the never reached if) or should I copy the functionality someplace else. I am asking because it was not written by me and I am not sure what should be done.

Reply:
I'm also not sure what needs to be done. I remember that isocontextualization was added for CERN so if it still works for you after removing and you see no need for the deleted functionality I'm fine with it.